pkg/engine/virtualmachinescalesets.go

```diff
@@ -312,7 +312,7 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 			AutoUpgradeMinorVersion: to.BoolPtr(true),
 			Settings:                map[string]interface{}{},
 			ProtectedSettings: map[string]interface{}{
-				"commandToExecute": fmt.Sprintf("[concat('retrycmd_if_failure() { r=$1; w=$2; t=$3; shift && shift && shift; for i in $(seq 1 $r); do timeout $t ${@}; [ $? -eq 0  ] && break || if [ $i -eq $r ]; then return 1; else sleep $w; fi; done }; "+outBoundCmd+" for i in $(seq 1 1200); do grep -Fq \"EOF\" /opt/azure/containers/provision.sh && break; if [ $i -eq 1200 ]; then exit 100; else sleep 1; fi; done; ', variables('provisionScriptParametersCommon'),%s,variables('provisionScriptParametersMaster'), ' /usr/bin/nohup /bin/bash -c \"/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1\"')]", generateUserAssignedIdentityClientIDParameter(userAssignedIDEnabled)),
+				"commandToExecute": fmt.Sprintf("[concat('echo $(date),$(hostname); retrycmd_if_failure() { r=$1; w=$2; t=$3; shift && shift && shift; for i in $(seq 1 $r); do timeout $t ${@}; [ $? -eq 0  ] && break || if [ $i -eq $r ]; then return 1; else sleep $w; fi; done }; "+outBoundCmd+" for i in $(seq 1 1200); do grep -Fq \"EOF\" /opt/azure/containers/provision.sh && break; if [ $i -eq 1200 ]; then exit 100; else sleep 1; fi; done; ', variables('provisionScriptParametersCommon'),%s,variables('provisionScriptParametersMaster'), ' /usr/bin/nohup /bin/bash -c \"/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1\"')]", generateUserAssignedIdentityClientIDParameter(userAssignedIDEnabled)),
```

Comment:
@PatrickLang @marosset is this something that would be useful as well for Windows VMs? If so, what's the equivalent for powershell that I could add to https://github.com/Azure/aks-engine/pull/1921/files#diff-438c69b891b52277ec1344112d40e26cR729?

Reply:
you could try replacing "commandToExecute": "[concat('powershell.exe -ExecutionPolicy Unrestricted -command with "commandToExecute": "[concat('**echo %TIME%,%COMPUTERNAME% &&** powershell.exe -ExecutionPolicy Unrestricted -command   below when the CSE gets run.
that should echo the current time and computer name before powershell.exe gets called to run the script.

Reply:
If you want the date too (based on the doc comments) i think you'll need to echo echo %DATE%,%TIME%,%COMPUTERNAME% 
That will output something similar to the following 
marosset@MAROSSET-WKS C:\Users\marosset
$ echo %DATE%,%TIME%,%COMPUTERNAME%
Wed 09/11/2019, 9:33:53.08,MAROSSET-WKS